addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1261,8 +1262,17 @@ const Wysiwyg = Widget.extend({
                         anchorOffset = focusOffset = index;
                     }
                 } else {
-                    anchorNode = link;
-                    focusNode = link;
+                    const selectionDirection = getCursorDirection(selection.anchorNode, 0, selection.focusNode, 0);
+                    if (
+                        closestElement(selection.anchorNode, 'a') === link &&
+                        closestElement(selection.focusNode, 'a') === link
+                    ) {
+                        [anchorNode, focusNode] = selectionDirection
+                            ? [selection.anchorNode, selection.focusNode]
+                            : [selection.focusNode, selection.anchorNode];
+                    } else {
+                        [anchorNode, focusNode] = [link, link];
+                    }
```

Comment:
About the test TBH I did not find a better place to add it, as `destroyLinkTools` is called when we click on side toolbar. What type of test case should I add a tour or an QUnit? And where in website or in mass_mailing? What do you suggest.

Reply:
There's a website tour for link tools (`link_tools.js`), maybe you could add it there?

Reply:
Note that it is currently disabled on runbot, mostly because of the ZWS introduction that broke it in different ways. We are still trying to make it possible to re-enable. Some things already merged but could you wait for those to be merged or closed (I think one or two were just for multi-build debugging) please? :pray: 
- https://github.com/odoo/odoo/pull/145770
- https://github.com/odoo/odoo/pull/144121
- https://github.com/odoo/odoo/pull/144234
- https://github.com/odoo/odoo/pull/144321

Thank you :pray:

Reply:
Of course, thanks for the info!